### PR TITLE
Add display_mode to assessments

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/views/course/assessment/assessments/_form.html.slim [hunk 1]
```diff
@@ -9,6 +9,7 @@
   = f.input :end_at
   = f.input :bonus_end_at
   = f.input :draft
+  = f.input :guided_mode, as: :select, collection: Course::Assessment.guided_modes.keys
```
Comment (allenwq) [inline app/views/course/assessment/assessments/_form.html.slim]: @lowjoel Think we should not allow changing this mode in `edit` ? what do you think ?

File: app/models/course/assessment.rb [hunk 2]
```diff
@@ -9,6 +9,8 @@ class Course::Assessment < ActiveRecord::Base
   before_validation :assign_folder_attributes
   before_validation :propagate_course
 
+  enum guided_mode: { normal: 0, step_by_step: 1 }
```
Comment (lowjoel) [inline app/models/course/assessment.rb]: I would call this 'presentation' and 'worksheet' vs 'guided'

File: app/views/course/assessment/assessments/_form.html.slim (hunk 1 above)
Comment (lowjoel) [inline app/views/course/assessment/assessments/_form.html.slim]: Can change it.

Comment (allenwq) [inline app/views/course/assessment/assessments/_form.html.slim]: So the thing is, we allow user to change from "training" to "mission" ? I think it should be freezed after creation ?

File: app/models/course/assessment.rb (hunk 2 above)
Comment (allenwq) [inline app/models/course/assessment.rb]: Okay, how about "display_mode" ?  think "presentation" is too general  ?

Comment (lowjoel) [inline app/models/course/assessment.rb]: okay that works

File: app/views/course/assessment/assessments/_form.html.slim (hunk 1 above)
Comment (allenwq) [inline app/views/course/assessment/assessments/_form.html.slim]: Okay, if it's just the presentation, think we should allow.

File: spec/features/course/assessment_management_spec.rb [hunk 3]
```diff
@@ -26,6 +26,9 @@
         fill_in 'assessment_start_at', with: assessment.start_at
         fill_in 'assessment_end_at', with: assessment.end_at
         fill_in 'assessment_bonus_end_at', with: assessment.bonus_end_at
+        within '#assessment_display_mode' do
+          find("option[value='guided']").select_option
```
Comment (lowjoel) [inline spec/features/course/assessment_management_spec.rb]: I think there's a `select` DSL, but this is fine.

**Pull request merged**